### DISCUSSION TITLE
HelloWorldExample on 2 different Machines

### DISCUSSION THREAD

**Discussion by ussefdesouky** [eProsima/Fast-DDS#3814] [Q&A]
How to run the HelloWorldExample that relies on the UDP protocol on 2 different devices connected on the same network, device1 publisher and device 2 subscriber.

Another question can 2 participants of different protocols pub and sub from each other or they must be using the same protocol.

**Comment by JLBuenoLopez** [comment]
@ussefdesouky

In order for two participants to communicate, the same transport layer must be used because each transport protocol follows its own specification.

Maybe you might consider contacting [eProsima commercial support team](https://github.com/eProsima/Fast-DDS#getting-help) in order to know the support options available for helping with your use case. I am afraid that all these questions (this one and others opened in the last few days) are out of the scope of the free support where the idea is to hint where in the documentation the information you are looking for can be found. If you want more specific information about how to use Fast DDS in your specific use case, then you will need to talk with eProsima's support team. Thanks for your understanding!
